test/Microsoft.AspNetCore.Razor.Evolution.Test/CaseSensitiveTagMatchingRuleComparer.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+using Microsoft.Extensions.Internal;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class CaseSensitiveTagMatchingRuleComparer : TagMatchingRuleComparer
```

Comment:
Also my opinion:

> Having multiple HtmlTargetElementAttributes with different cases may pass on the product variant of TagMatchingRuleComparer but not the case sensitive one depending on if we consume the attributes correctly.

Equal means Equal, and equal is defined by the default comparer. Having multiple values that vary by only case is the the same as having multiple values that are literally the same unicode code points. We should either ignore duplicates everywhere or report an error everywhere. Anything else is a bug.